### ISSUE TITLE
85ef1c0 breaks AS5114-48X : Driver is incomatible with FW

### ISSUE THREAD

**Issue by aep** [dentproject/dentOS#147]
85ef1c0 breaks accton-as5114-48x and possibly all the boards with the same asic.

**Comment by aep** [comment]
```
[   25.297438] sfp sfp-43: Host maximum power 2.0W
[   25.303934] libphy: SFP I2C Bus: probed
[   25.308007] sfp sfp-44: Host maximum power 2.0W
[   25.310049] Loading Marvell Prestera Switch Driver
[   25.317412] libphy: SFP I2C Bus: probed
[   25.321456] sfp sfp-45: Host maximum power 2.0W
[   25.329550] libphy: SFP I2C Bus: probed
[   25.333628] sfp sfp-46: Host maximum power 2.0W
[   25.349674] libphy: SFP I2C Bus: probed
[   25.353721] sfp sfp-47: Host maximum power 2.0W
[   25.355969] Aldrin2 0000:01:00.0: FW version '2.8.0'
[   25.363280] Aldrin2 0000:01:00.0: Driver version '3.0.1'
[   25.368622] Aldrin2 0000:01:00.0: Driver is incomatible with FW: version mismatch
[   25.376144] Aldrin2 0000:01:00.0: FW image is invalid
[   25.377794] Aldrin2: probe of 0000:01:00.0 failed with error -22
[   25.381244] libphy: SFP I2C Bus: probed
[   25.387354] prestera_pci: Registered Marvell Prestera PCI driver
[   25.397359] sfp sfp-48: Host maximum power 2.0W
Setting up base platform configuration for arm64-accton-as5114-48x-r0: done
INFO:PKI:Using existing private key.
```

**Comment by paulmenzel** [comment]
Thank you for reporting the issue. So GitHub links to the commit: That is commit 85ef1c058 (Update prestera driver to v3.0.0 (#133)). Maybe edit the first message.

**Comment by paulmenzel** [comment]
In the linux-firmware repository I found the file `mrvl/prestera/mvsw_prestera_fw-v3.0.img` added in [commit 16052e4a (mrvl: prestera: Add Marvell Prestera Switchdev firmware 3.0 version)](https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/commit/?id=16052e4a41bd0d47793fa33eb718081df28efe11).

> Main changes:  
>   
>     - add initial L3 support  
>     - changed a bit the way of events sending  
>     - changed ABI for some IPC calls  
>     - add LAG support  
>   
> This firmware version is ABI un-compatible with the current 2.0,  
> so add it as separate binary.

It’s not the real solution, but could you please test, if that file works for you?

In my opinion, it violates Linux’ no-regression policy if the ASIC stops working with a newer Linux kernel and old firmware version.

**Comment by aep** [comment]
sure. unfortunately one build of dentos takes 3 days at 25kb/s but i'll try.

**Comment by paulmenzel** [comment]
I think, copying that file to the directory `/lib/firmware/mrvl/prestera/` should be enough.

**Comment by robimarko** [comment]
That FW from linux-firmware wont work.
You need the 3.0.1 FW from here:
https://github.com/dentproject/dent-artifacts/blob/master/REPO/stretch/packages/binary-arm64/mrvl-fw-image_3.0.1_arm64.deb

The pointer has been updated to 3.0.0 FW in the DENT, but not to 3.0.1.
Have you updated your git submodules in DENT repo?

**Comment by aep** [comment]
> You need the 3.0.1 FW from here:

can confirm this works

> Have you updated your git submodules in DENT repo?

no. this updates to 3.0, which you say doesnt work?

**Comment by robimarko** [comment]
3.0 in DENT and linux-firmware arent the same at all, but yeah it won't work as the driver requires 3.0.1 and the pointer wasn't updated along with the driver.

**Comment by aep** [comment]
i tested 3.0.1 a while and there's several fatal defects.
However, i'm not sure if i should report them at all, since it seems unlikely marvel will even look at this github?

**Comment by sonoble** [comment]
They will look. Please report.

**Comment by taraschornyiplv** [comment]
fixed by #144
